lib/chef/provider/package/yum/python_helper.rb

```diff
@@ -40,9 +40,16 @@ class PythonHelper
           YUM_HELPER = ::File.expand_path(::File.join(::File.dirname(__FILE__), "yum_helper.py")).freeze
 
           def yum_command
-            @yum_command ||= which("platform-python", "python", "python2", "python2.7", extra_path: "/usr/libexec") do |f|
-              shell_out("#{f} -c 'import yum'").exitstatus == 0
-            end + " #{YUM_HELPER}"
+            @yum_command ||= begin
+              cmd = which("platform-python", "python", "python2", "python2.7", extra_path: "/usr/libexec") do |f|
+                res = shell_out("#{f} -c 'import yum'")
+                raise res.stderr if res.exitstatus != 0
+
+                true
+              end
+
+              "#{cmd} #{YUM_HELPER}"
+            end
```

Comment:
So I think what I'd like better here would look more like:

```ruby
@yum_command ||= begin
    cmd = which("platform-python", "python", "python2", "python2.7", extra_path: "/usr/libexec") do |f|
              shell_out("#{f} -c 'import yum'").exitstatus == 0
            end
    raise Chef::Exceptions::Package, "cannot find yum libraries, you may need to use dnf_package" unless cmd
    "#{cmd} #{YUM_HELPER}"
  end
```

The code here in this PR has a bug where it'll only try to the first python it finds and if that doesn't have the yum/dnf libraries it'll throw.  It needs to preserve the behavior of searching through all the python executables that it finds to find the one that has the yum/dnf libs.

If you want to go the extra mile you could Chef::Log.trace log the stderr of all the shell_out's that are executed (I don't think users will generally want that output)

Reply:
@lamont-granquist Updated the required changes. Thankyou!